guides/source/sign_up.md

```diff
@@ -0,0 +1,1652 @@
+  </div>
+
+  <div>
+    <%= form.label :password %>
+    <%= form.password_field :password, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password_confirmation %>
+    <%= form.password_field :password_confirmation, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.submit "Sign up" %>
+  </div>
+<% end %>
+```
+
+This form collects the user's name, email, and password. We're using the `autocomplete` attribute to suggest to the browser the content requested for these fields.
+
+You'll also notice we set `url: sign_up_path` in the form alongside `model: @user`. Without the `url:` argument, `form_with` would see we have a `User` and send the form to `/users`. Since we want the form to submit to `/sign_up`, the `url:` argument tells the form exactly where to submit the data.
+
+Back in `app/controllers/sign_ups_controller.rb` we can handle the form submission by adding the `create` action.
+
+```ruby#6-19
+class SignUpsController < ApplicationController
+  def show
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(sign_up_params)
+    if @user.save
+      start_new_session_for(@user)
+      redirect_to root_path
+    else
+      render :show, status: :unprocessable_entity
+    end
+  end
+
+  private
+    def sign_up_params
+      params.expect(user: [ :first_name, :last_name, :email_address, :password, :password_confirmation ])
+    end
+end
+```
+
+The `create` action assign parameters and attempts to save the user to the database. If successful, it logs the user in and redirects to `root_path`, otherwise it re-renders the form with errors.
+
+One last fix we need to make is restricting `SignUpsController` to unauthenticated users only. Let's do that by adding a helper to the `Authentication` module in `app/controllers/concerns/authentication.rb`.
+
+```ruby#14-17
+module Authentication
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :require_authentication
+    helper_method :authenticated?
+  end
+
+  class_methods do
+    def allow_unauthenticated_access(**options)
+      skip_before_action :require_authentication, **options
+    end
+
+    def unauthenticated_access_only(**options)
+      allow_unauthenticated_access **options
+      before_action -> { redirect_to root_path if authenticated? }, **options
+    end
+
+    # ...
+```
+
+The `unauthenticated_only_access` class method can be used in any controller where we want to restrict actions to unauthenticated users only.
+
+We can then use this method at the top of `SignUpsController`.
+
+```ruby#2
+class SignUpsController < ApplicationController
+  unauthenticated_access_only
+
+  # ...
+end
+```
+
+### Rate Limiting Sign Up
+
+Our application will be accessible on the Internet so we're bound to have malicious bots and users trying to spam our application. We can add rate limiting to sign up to slow down anyone submitting too many requests.
```

Comment:
```suggestion
Our application will be accessible on the internet so we're bound to have malicious bots and users trying to spam our application. We can add rate limiting to sign up to slow down anyone submitting too many requests.
```

If there's a reason to keep internet capitalized let me know! Otherwise, lower case.